lib/validation.js

```diff
@@ -73,11 +73,23 @@ function validateParam (validatorFunction, request, paramName) {
 }
 
 function validate (context, request) {
-  return validateParam(context[paramsSchema], request, 'params') ||
-    validateParam(context[bodySchema], request, 'body') ||
-    validateParam(context[querystringSchema], request, 'query') ||
-    validateParam(context[headersSchema], request, 'headers') ||
-    true
+  var params = validateParam(context[paramsSchema], request, 'params')
+  if (params) {
+    return { cause: 'params', valid: params }
```

Comment:
I would suggest creating the error here and returning it instead of using `wrapValidationError` in `handleRequest.js` (maybe move `wrapValidationError` into this file and use it here).

Reply:
Good point.

Reply:
Done.